prestoadmin/package.py

```diff
@@ -18,21 +18,23 @@
 import logging
 
 from fabric.context_managers import settings, hide, shell_env
-from fabric.decorators import task
+from fabric.decorators import task, runs_once
 from fabric.operations import sudo, put, os, local
 from fabric.state import env
+from fabric.tasks import execute
 from fabric.utils import abort
 
 from prestoadmin.util import constants
 from prestoadmin.standalone.config import StandaloneConfig
 from prestoadmin.util.base_config import requires_config
-
+from prestoadmin.util.fabricapi import get_host_list
 
 _LOGGER = logging.getLogger(__name__)
 __all__ = ['install']
 
 
 @task
+@runs_once
 @requires_config(StandaloneConfig)
 def install(local_path):
```

Comment:
I think you probably also need to add check_if_valid_rpm to server.py's install method.